runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -420,7 +425,7 @@ public CorfuRuntime registerBeforeRpcHandler(Runnable handler) {
                         try {
                             newRouter.addClient(new LayoutClient())
                                 .addClient(new SequencerClient())
-                                .addClient(new LogUnitClient().setMetricRegistry(metrics != null
+                                .addClient(new LogUnitClient(getParameters()).setMetricRegistry(metrics != null
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)